skiko/src/awtMain/objectiveC/macos/MetalRedrawer.mm

```diff
@@ -92,12 +113,24 @@ JNIEXPORT jlong JNICALL Java_org_jetbrains_skiko_redrawer_MetalRedrawer_createMe
         /// max inflight command buffers count matches swapchain size to avoid overcommitment
         device.inflightSemaphore = dispatch_semaphore_create(device.layer.maximumDrawableCount);
 
-        if (transparency)
-        {
-            NSWindow* window = (__bridge NSWindow*) (void *) windowPtr;
+        NSWindow* window = (__bridge NSWindow*) (void *) windowPtr;
+
+        if (transparency) {
             window.hasShadow = NO;
         }
 
+        jclass redrawerClass = env->GetObjectClass(redrawer);
+        jmethodID onOcclusionStateChanged = env->GetMethodID(redrawerClass, "onOcclusionStateChanged", "(Z)V");
+
+        [[NSNotificationCenter defaultCenter] addObserverForName:NSWindowDidChangeOcclusionStateNotification
+                                                          object:window
+                                                           queue:[NSOperationQueue mainQueue]
+                                                      usingBlock:^(NSNotification * _Nonnull note) {
+            BOOL isOccluded = ([window occlusionState] & NSWindowOcclusionStateVisible) == 0;
+            JNIEnv *jniEnv = resolveJNIEnvForCurrentThread();
+            jniEnv->CallObjectMethod(layer.javaRef, onOcclusionStateChanged, isOccluded);
+        }];
```

Comment:
This method returns a handle, which must be used to deregister this observation once it's not needed.
https://developer.apple.com/documentation/foundation/nsnotificationcenter/1411723-addobserverforname

Reply:
Fixed (hopefully)